lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'license_detector.dart';
+
+@sealed
+class License {
+  /// SPDX identifier of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// Map from [Token.value] to their number of occurrences in this license.
+  final Map<String, int> occurrences;
+
+  License._(this.content, this.tokens, this.occurrences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
```

Comment:
You can make this easier to use by using named params:
```suggestion
  factory License.parse({required String identifier, required String content}) {
```